### ISSUE TITLE
"Run Tests" doesn't point to the right dir with `shinytest`

### ISSUE THREAD

**Issue by trestletech** [rstudio/rstudio#5677]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.423
    OS Version      : Mac OSX
    R Version       : 3.5.1

### Steps to reproduce the problem

 - install shinytest from CRAN
 - Record a Shiny test for a Shiny application, leave "Open script in editor on exit" checked.
 - When the script opens, click the "Run Tests" button
 - Observe the following error:

```
==> Testing Shiny application using 'shinytest'

Error in shinytest::testApp("~/Temp/testedapp/testy/tests", "mytest.R") : 
  Test scripts do not exist: mytest.R
In addition: Warning message:
package ‘shinytest’ was built under R version 3.5.2 
Execution halted

Exited with status 1.

```

### Describe the problem in detail

The problem is that shinytest -- with the last CRAN release -- moved to nesting the tests two levels deep instead of one. So now the script that gets opened is in `tests/shinytest/` rather than the top-level `tests/` directory as happened in the past.

I believe the IDE is improperly guessing that the app dir is just one level up from the script, but it's now variable as we support having the test scripts in `tests/*.R` or in `tests/shinytest/*.R` -- but recommend and default to the latter.

We've opened an issue in shinytest to work around this for the time being (https://github.com/rstudio/shinytest/issues/293) but long-term we should update the IDE to make sure that it's pointing to the right appdir when you click "Run Tests"

### Describe the behavior you expected

"Run tests" would properly find the app directory and run the tests as it did when apps had their shinytest scripts in `tests/*.R` -- regardless of whether the tests are nested two levels deep (the new normal) or one level (the old way).

**Comment by dfalty** [comment]
Hey @trestletech I actually can't reproduce this with the latest `shinytest` CRAN release (1.3.1). I still see the one level nesting of `tests/` with both IDE versions `1.3.567` & `1.2.5019`. Am I missing something here?

**Comment by trestletech** [comment]
Oops. Sorry. Turns out we didn't release this change to CRAN. And we just merged the workaround into shinytest, so you shouldn't be able to repro on the latest of master.

To reproduce the issue, you'd need to

1. Delete your `tests/` directory (This might be required I don't remember, so let's go ahead and do it.)
2. run `remotes::install_github("rstudio/shinytest", ref="81eeaed223a0ccf3fbff6d30a51955e07e40550e")`
3. Restart the session, just to make sure you're using the new shinytest package version
4. Record a test using `shinytest::recordTest()`
5. Click "Run Tests"

Here's a jif:

![shinytest](https://user-images.githubusercontent.com/1593639/67997992-98a35380-fc4e-11e9-94be-bd5817ebf5e8.gif)

But given that we a.) didn't release this change to CRAN, and b.) already have a workaround on master of shinytest, I think it's safe to call this issue low priority. I would argue that we should still keep it open given that the IDE's shinytest integration is pointing to the wrong `appDir` when it calls `testApp()`, but given that there's no user-visible impact, I think it could safely wait.

**Comment by trestletech** [comment]
And just so the issues are linked.... fixed in https://github.com/rstudio/shinytest/pull/294

**Comment by dfalty** [comment]
Ah, that makes sense. Thanks for the additional information.

**Comment by trestletech** [comment]
shinytests' test-dir-finder is here: https://github.com/rstudio/shinytest/blob/master/R/test-app.R#L84-L99

**Comment by jmcphers** [comment]
This blocks work the shinytest team wants to demo at the conference, so it'd be great to get it into Orange Blossom (which should be GA by then)

**Comment by dfalty** [comment]
Verified `Run Tests` works correctly with the specific version of `shinytest` listed in the bug on RStudio version 1.2.5031. 

1. remotes::install_github("rstudio/shinytest", ref="81eeaed223a0ccf3fbff6d30a51955e07e40550e")
2. Record Test
3. Save Test
4. Run Test using `Run Tests` button
5. Observe test runs correctly

BUT, it looks like the `Compare Results` button needs to be hooked up similarly because it is not working correctly.

6. Edit test to something that intentionally fails.
7. Save and Run Tests
8. Observe test fails
9. Select Compare Results
10. Observe RStudio reports `No Failed Results` even though there are failed tests to compare. This works from `shinytest::snapshotCompare()`
![image](https://user-images.githubusercontent.com/15237845/69985868-b5f36800-1501-11ea-94da-53b5bfe04ac0.png)

**Comment by dfalty** [comment]
Verified fixed 1.2.5033-1
